procbuild/builder.py

```diff
@@ -77,99 +76,188 @@ def shell(cmd, path=None, retry=0):
+                         'build_output': self.build_output,
+                         'build_pdf_path': self.build_pdf_path,
+                         'build_timestamp': self.build_timestamp}
+                }
+
+
+    def _get_build_tools(self):
+        """This command will get the latest version of the repo path.
+        
+        """
+        if not os.path.exists(self.master_repo_path):
+            self.add_output('[*] Checking out proceedings build tools '
+                            f'to {self.master_repo_path}...\n')
+            errcode, output = checkout(repo('scipy-conference'), 
+                                       self.master_branch,
+                                       self.master_repo_path)
+        else:
+            self.add_output('[*] Updating proceedings build tools in'
+                            ' {self.master_repo_path}...\n')
+            errcode, output = shell('git pull', self.master_repo_path, retry=2)
+        
+        self.add_output(output)
+        
+        if errcode:
+            self.add_output('[X] Error code %d ' % errcode)
+            raise BuildError("get_build_tools")
+        
+        self._remove_papers_dir()
+        
+    
+    def _remove_papers_dir(self):
+        """ this command will remove the papers dir from the build_tools dir
+        
+        """
+        errcode, output = shell('rm -rf papers', 
+                                path=self.master_repo_path)
+        
+        self.add_output(output)
+
+        if errcode:
+            self.add_output('[X] Error code %d ' % errcode)
+            raise BuildError("remove_papers_dir")
+
+    def _checkout_paper_repo(self):
+        self.add_output('[*] Check out paper repository...\n')
+        errcode, output = checkout(repo(self.user), 
+                                   self.branch, 
+                                   self.build_path)
+        self.add_output(output)
+        
+        if errcode:
+            self.add_output('[X] Error code %d\n' % errcode)
+            self.build_status = 'Failed to check out paper'
+            raise BuildError('checkout_paper_repo')
+
+    
+    def _relocate_build_tools(self):
+        """Move local build tools into temporary directory
+        """
+        self.add_output('Moving proceedings build tools to temp directory...\n')
+        errcode, output = shell('cp -r '
+                                f'{self.master_repo_path}/. {self.build_path}')
+        self.add_output(output)
+        if errcode:
+            self.add_output('[X] Error code %d\n' % errcode)
+            self.build_status = 'Could not move build tools to temp directory'
+            raise BuildError('relocate_build_tools')
+
+    def _relocate_static_files(self):
+        for f in self.data_files:
+            shutil.copy(f, self.paper_path)
+    
+    @property
+    def paper_path(self):
+        return joinp(self.build_path, 'papers', self.paper)
+        
+    @property
+    def paper(self):
```

Comment:
What does this method do?  Naming doesn't guide the reader, and no docstring.

Reply:
I see now how this works.  Hrm, wondering if it makes sense to have this at all, or whether it could just be a parameter to the `build_paper` method?

Reply:
Making it a parameter would require that we do the parsing of the PR outside of the Build manager, right now all of that parsing is being done in here. If we were going to switch that, I would want to include that functionality in the `PRManager`, which is a bigger refactor. Is this fine for now?

Reply:
In the meantime I can add a docstring